presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/StateCompiler.java

```diff
@@ -225,7 +230,7 @@ private static void generateGetSerializedType(ClassDefinition definition, List<S
                         .ifFalse(state.cast(setter.getDeclaringClass()).invoke(setter, constantType(binder, field.getSqlType()).getValue(block, index))));
             }
             else {
-                // For primitive type, we need to cast here because we serialize byte fields with TINYINT (whose java type is long).
+                // For primitive type, we need to cast here because we serialize byte fields with TINYINT/INTEGER (whose java type is long).
```

Comment:
what about `SMALLINT` (`short`)?

Reply:
Looks like for now we don't have a need to serialize/deserialize a 2-byte data type (e.g., `char`) in any of our state. We can add them later on if necessary like `TriStateBooleanState` or the new states in this PR.